### PR TITLE
Search on landing

### PR DESCRIPTION
**Pull request by harrysolovay**
Move search to universal nav. Add different mobile treatments for landing & other pages.

### CODE REVIEW

File: client/src/utils/hyperscript/index.ts [hunk 1]
```diff
@@ -1,17 +1,11 @@
 import {h} from "@stencil/core";
 import {VNode} from "@stencil/core";
-import {isHTMLTagName} from "is-html-tag-name";
```
Comment (jordanranz) [inline client/src/utils/hyperscript/index.ts:3]: Are these removals related to this PR?

File: client/src/docs-ui/search-bar/search-bar.style.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import {MQLaptop} from "../../amplify-ui/styles/media";
+import {css} from "emotion";
+
+// we use !important to override webkit's imposed border-radius
+export const searchStyle = css`
+  width: 100%;
+
+  ${MQLaptop} {
+    width: initial;
+    margin: 0 1rem;
```
Comment (jordanranz) [inline client/src/docs-ui/search-bar/search-bar.style.ts:10]: `margin: 0 4rem` looks a little better. Let's chat on this tomorrow.

**Comment by jordanranz** [comment]
The header in the algolia search results is missing for some reason when you search from inside the site (not landing page).

File: client/src/utils/hyperscript/index.ts (hunk 1 above)
Comment (harrysolovay) [reply client/src/utils/hyperscript/index.ts:3]: No, but it's good housekeeping

**Comment by jpignata** [comment]
Random design thought: can we show the search in the sticky sub nav when the top nav scrolls off the page?

**Comment by jpignata** [comment]
As @jordanranz mentioned, I don't see the header for each result group inside the site. I'm using Chrome 83.0.4103.61.

CleanShot 2020-06-04 at 14 07 01

File: client/src/docs-ui/search-bar/search-bar.style.ts (hunk 2 above)
Comment (sammartinez) [reply client/src/docs-ui/search-bar/search-bar.style.ts:10]: In agreement with Jordan on this, can we make the change?

File: client/src/docs-ui/search-bar/search-bar.style.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        color: black;
+      }
+
+      .algolia-autocomplete .algolia-docsearch-suggestion--text {
+        font-size: 0.8rem;
+        color: gray;
+      }
+
+      .ds-cursor .algolia-docsearch-suggestion--content {
+        background-color: var(--bg-color-hover) !important;
```
Comment (sammartinez) [inline client/src/docs-ui/search-bar/search-bar.style.ts:64]: Is there another option we can use in order to avoid using `!important`?

File: client/src/docs-ui/search-bar/search-bar.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import {searchStyle} from "./search-bar.style";
+import {transformData} from "../../utils/transform-search-data";
+
+@Component({tag: "docs-search-bar", shadow: false})
+export class DocsSearchBar {
+  initDocSearch() {
+    if (Build.isBrowser) {
+      // @ts-ignore
+      docsearch({
+        apiKey: "24d37f059982b2f5ecf829afe93aed40",
```
Comment (sammartinez) [inline client/src/docs-ui/search-bar/search-bar.tsx:11]: Is there a way we can adjust this to be a CONSTANT? Or is this going to change over time?

File: client/src/docs-ui/secondary-nav/secondary-nav.tsx [hunk 5]
```diff
@@ -1,33 +1,27 @@
-import {Component, Host, h, Prop, Build} from "@stencil/core";
+import * as links from "../../constants/links";
```
Comment (sammartinez) [inline client/src/docs-ui/secondary-nav/secondary-nav.tsx:1]: NIT/NON-BLOCKING: I feel it would make more sense to import each individual constant. What was the reason on this?

File: client/src/utils/hyperscript/index.ts (hunk 1 above)
Comment (sammartinez) [reply client/src/utils/hyperscript/index.ts:3]: I like the house cleaning, we may want to push it to a separate PR in case it leads to different regressions with it.

File: client/src/docs-ui/secondary-nav/secondary-nav.tsx (hunk 5 above)
Comment (harrysolovay) [reply client/src/docs-ui/secondary-nav/secondary-nav.tsx:1]: It's less writing / more readable imo

File: client/src/docs-ui/search-bar/search-bar.tsx (hunk 4 above)
Comment (harrysolovay) [reply client/src/docs-ui/search-bar/search-bar.tsx:11]: Fair enough––will move into constants

File: client/src/docs-ui/search-bar/search-bar.style.ts (hunk 3 above)
Comment (harrysolovay) [reply client/src/docs-ui/search-bar/search-bar.style.ts:64]: I commented above that style. Algolia's styles were a pain to override. Time-boxed it and moved on. I urge you to try... although this one's a tricky beast! It's tough––in this case––to get a more specific selector, which accounts for conflict with algolia-injected styles.

Comment (harrysolovay) [reply client/src/docs-ui/search-bar/search-bar.style.ts:64]: & yea––will make that change

**Comment by jpignata** [comment]
These headers don't make much sense.

CleanShot 2020-06-04 at 19 56 20@2x

**Documentation (Ionic)** vs. **Libraries (iOS)**? Is the iOS section not documentation?

CleanShot 2020-06-04 at 19 55 52@2x

**Documentation (JavaScript)** vs. **CLI**? Same question.

**Comment by jpignata** [comment]
CleanShot 2020-06-04 at 19 59 00@2x

Some formatting issues at narrower sizes.

**Comment by harrysolovay** [comment]
@jpignata agreed –– but I don't think that can be addressed in this PR. It's a matter of fixing the index (again ;). Aside from the heading text, everything looking good?

File: client/src/amplify-ui/styles/media.ts [hunk 6]
```diff
@@ -21,3 +21,6 @@ export const MQDesktop = createDeviceMQ(Breakpoint.DESKTOP);
 export const MQMonitor = createDeviceMQ(Breakpoint.MONITOR);
 
 export const MAX_WIDTH = "90rem";
+
+export const isWiderThanTablet = () =>
```
Comment (jpignata) [inline client/src/amplify-ui/styles/media.ts:25]: For your consideration: This file is a little confusing to me. What's the difference between `MONITOR` and `DESKTOP` for example, and `isWiderThanTablet` is kind of confusing to me given tablets operate in portrait and landscape, with vastly different aspect ratios.

**Comment by jpignata** [comment]
This is production, so probably not a regression.

CleanShot 2020-06-04 at 20 12 31@2x

File: client/src/docs-ui/hero/hero.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import {heroStyle, landingSearchStyle} from "./hero.style";
+import {isWiderThanTablet} from "../../amplify-ui/styles/media";
+
+@Component({tag: "docs-hero", shadow: false})
+export class DocsHero {
+  @State() isWiderThanTablet = this.calculateIsWiderThanTablet();
+
+  @Listen("resize", {target: "window"})
+  calculateIsWiderThanTablet() {
+    this.isWiderThanTablet = isWiderThanTablet();
```
Comment (jpignata) [inline client/src/docs-ui/hero/hero.tsx:11]: Nit: Why not name this expressively for its purpose? In this case, it's `showSearch` or something similar. On L21, it's hard to even see the bang, so it might benefit from more purposeful naming.

File: client/src/utils/hyperscript/index.ts (hunk 1 above)
Comment (jpignata) [reply client/src/utils/hyperscript/index.ts:3]: :+1: – I'd move this into a separate PR.

File: amplify/.config/project-config.json [hunk 8]
```diff
@@ -1,17 +0,0 @@
-{
```
[the remark this answers is not in the document]
Comment (harrysolovay) [reply amplify/.config/project-config.json:1]: Thank you, thank you, thank you for catching that. Could have been COE.

File: client/src/amplify-ui/styles/media.ts (hunk 6 above)
Comment (harrysolovay) [reply client/src/amplify-ui/styles/media.ts:25]: You're correct! In that case, we should rename the properties of the `Breakpoint` enum.

File: amplify/.config/project-config.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (jordanranz) [reply amplify/.config/project-config.json:1]: It would not have taken anything down, this folder is mainly used for future updates to the cloud stack.
